src/api/app/models/project.rb

```diff
@@ -1245,14 +1246,34 @@ def update_product_autopackages
   end
 
   def open_requests
-    reviews = BsRequest.collection(project: name, states: %w(review)).map(&:number)
-    targets = BsRequest.collection(project: name, states: %w(new)).map(&:number)
-    incidents = BsRequest.collection(project: name, states: %w(new), types: %w(maintenance_incident)).map(&:number)
-
+    reviews = BsRequest.with_involved_projects(id)
+                       .or(BsRequest.with_source_projects(id))
+                       .with_actions_and_reviews
+                       .where(state: :review, reviews: { state: :new })
+                       .pluck(:number)
+
+    targets = BsRequest.with_involved_projects(id)
+                       .or(BsRequest.with_source_projects(id))
+                       .in_states(:new).with_actions
+                       .pluck(:number)
+
+    incidents = BsRequest.with_involved_projects(id)
+                         .or(BsRequest.with_source_projects(id))
+                         .in_states(:new)
+                         .with_types(:maintenance_incident)
+                         .with_actions
+                         .pluck(:number)
+
+    maintenance_release = []
     if is_maintenance?
-      maintenance_release = BsRequest.collection(project: name, states: %w(new), types: %w(maintenance_release), subprojects: true).map(&:number)
-    else
-      maintenance_release = []
+      project_pattern_name = name + ':%'
+
+      maintenance_release = BsRequest.with_target_subprojects(project_pattern_name)
```

Comment:
True, source project and/or package can be remote.

Target project and package may also not exist, if it is either new package or an old request and project and/or package got removed meanwhile.